### PR TITLE
Updates for frontend2

### PR DESCRIPTION
**Pull request by wenqing**
as titled.

Besides, changed the name of the benchmark of `MPI/hm_tri/hm_tri` to `MPI/HM/hm`.

### CODE REVIEW

**Comment by wenqing** [comment]
@bilke Please take a look. Or you can send a PR for the changes in the modules.

**Comment by wenqing** [comment]
@bilke I dropped the part to rename envinf1 to frontend.

**Comment by wenqing** [comment]
@bilke Thanks. I will check the results of the benchmarks that use MKL and LIS.

**Comment by bilke** [comment]
@wenqing LIS is green, MKL has a failing benchmark.

**Comment by wenqing** [comment]
@bilke There is a runtime error MKL FATAL ERROR: Cannot load libmkl_lapack.so. libmkl_lapack.so is not included in that precompiled lib. On envinf1, it is in /opt/intel/mkl. I will try to fix this runtime error. If there is no solution for that, I will remove the benchmarks for MKL temporarily,

**Comment by wenqing** [comment]
MKL benchmarks are disabled in ctest because one MKL library is missed on the frontends of eve.

File: FEM/matrix_routines.cpp [hunk 1]
```diff
@@ -3162,6 +3162,9 @@ void M1Vorkond(int aufgabe, double* x, double* b)
                         diag -= dim + 1;
                     } /* i */
                 }     /* Modus iLDU, (zerlegen und) aufloesen */
+#ifdef GCC
```
Comment (wenqing) [inline FEM/matrix_routines.cpp]: Fall through warnings leads to an unstable warning error on Jenkins.

**Comment by wenqing** [comment]
Deactivated the benchmark of  AKS/GAS_FLOW/Gravity/Gravity, which sporadically failed or succeeded in linear solver convergence in the code compiled with gcc7.

The benchmark  uses the exponential fluid density model (type 14), which has a bug. With the bugfixing in  the exponential fluid density model, the  sporadically failure still exists. Valgrind gives no memory leakage in running this benchmark. The benchmark runs good with other density model.

**Comment by wenqing** [comment]
I am going to merge it soon.

File: FEM/matrix_routines.cpp [hunk 2]
```diff
@@ -3347,7 +3348,7 @@ void M34Vorkond(int aufgabe, double* x, double* b)
             u = 0;
             o = w->usym;
             /* kein break! "Fall through" nach Aufgabe 2 */
-
+            // fall through
```
Comment (wenqing) [inline FEM/matrix_routines.cpp]: Maker comment to avoid "fall through" warning.

**Pull request merged**